packages/graphql_server/src/starHistory/issueHistory.ts

```diff
@@ -0,0 +1,302 @@
+import axios, { AxiosResponse } from 'axios'
+import { IssueRecord, IssueData, TimeFrame } from './types'
+import * as utils from './utils'
+
+const DEFAULT_PER_PAGE = 30
+
+/** This method retrieves the issues from a page of a GitHub repository
+ * @param {string} repo - Name of the Github repository: "owner/repository"
+ * @param {string} token - Github Access token
+ * @param {number} page - Page Number of the issues to retrieve
+ * @param {string} direction - Determines in which order the pages will be: 'asc' | 'desc'
+ * All the normal history functions should use 'asc', the partial history should use 'desc'
+ * @returns Promise<Object>: A promise that resolves to the issues of the repository
+ */
+async function getRepoIssues(
+  repo: string,
+  token: string,
+  page?: number,
+  direction?: string
+): Promise<AxiosResponse<IssueData[]>> {
+  let url = `https://api.github.com/repos/${repo}/issues?per_page=${DEFAULT_PER_PAGE}`
+
+  if (direction != undefined) {
+    url += `&sort=created&direction=${direction}`
+  }
+
+  if (page !== undefined) {
+    url = `${url}&page=${page}`
+  }
+  return axios.get(url, {
+    headers: {
+      Accept: 'application/vnd.github+json',
+      Authorization: token ? `token ${token}` : ''
+    }
+  })
+}
+
+/** Retrieves the total count of issues for a Github repository
+ * @param {string} repo - Name of the Github repository: "owner/repository"
+ * @param {string} token - Github access token
+ * @returns Promise<number> : A promise that resolves to the total count of issues for the repository
+ */
+async function getRepoIssuesCount(repo: string, token?: string): Promise<number> {
+  const response: AxiosResponse<{ open_issues_count: number }> = await axios.get(
+    `https://api.github.com/repos/${repo}`,
+    {
+      headers: {
+        Accept: 'application/vnd.github+json',
+        Authorization: token ? `token ${token}` : ''
+      }
+    }
+  )
+
+  return response?.data?.open_issues_count
+}
+
+/** Retrieves the issue records (issue count by date) of a Github repository
+ * @param {string} repo - Name of the Github repository: "owner/repository"
+ * @param {string} token - Github Access token
+ * @param {number} maxRequestAmount - Maximum number of API requests to make to retrieve the issues
+ * The higher this value is the more accurate is going to be the graph of the issue history
+ * @param {number} startPage - a possible start page for the partial history
+ * @returns Promise<Array<{ date: string, count: number }>>: A promise that resolves to an array of issue records
+ */
+export async function getRepoIssueRecords(
+  repo: string,
+  token: string,
+  maxRequestAmount: number,
+  startPage?: number
+) {
+  const pageCount = await getPageCount(repo, token)
+
+  const requestPages: number[] = []
+  if (startPage == undefined) {
```

Comment:
You check undefined, but you don’t check null please check null or just use isNullorUndefined function. I am not sure it is written. If it is not, you can add this